File: C#/InterviewCampProblems/InterviewCampProblems.Test/UnitTest3.cs
/*
Given a sentence, reverse the words of the sentence.For example,"i live in a house" becomes "house a in live i"
*/

using System;
using InterviewCampProblems.Code;
using Xunit;

namespace InterviewCampProblems.Test {
    public class UnitTest3 {
        [Fact]
        public void Test1 () {
            string expected = "house a in live i";
            string sentence = "i live in a house";
            var actual = Class3.ReverseSentence (sentence);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test2 () {
            string expected = " ";
            string sentence = " ";
            var actual = Class3.ReverseSentence (sentence);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test3 () {
            string expected = null;
            string sentence = null;
            var actual = Class3.ReverseSentence (sentence);
            Assert.Equal (expected, actual);
        }

    }
}

File: Java/InterviewCamp/src/test/java/UnitTest3.java
//Given a sentence, reverse the words of the sentence.For example,"i live in a house" becomes "house a in live i"

import org.junit.Assert;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertSame;

public class UnitTest3 {

    @Test
    public void When_Input_Correct() {

        //Arrange
        String str = "i live in a house";
        String outputstr = "house a in live i";
        //Act
        String output = Class3.ReverseWordsOfStrings(str);

        //Assert
        Assert.assertEquals(output,outputstr);

    }
}